haystack/preview/components/audio/transcriber.py

```diff
@@ -0,0 +1,209 @@
+from typing import List, Optional, Dict, Any, Union, BinaryIO, Literal, get_args, Sequence
+
+import os
+import json
+import logging
+from pathlib import Path
+from dataclasses import dataclass
+
+import requests
+import torch
+import whisper
+from tenacity import retry, wait_exponential, retry_if_not_result
+
+from haystack.preview import component, Document
+from haystack.errors import OpenAIError, OpenAIRateLimitError
+from haystack import is_imported
+
+
+logger = logging.getLogger(__name__)
+
+
+OPENAI_TIMEOUT = float(os.environ.get("HAYSTACK_OPENAI_TIMEOUT_SEC", 30))
+
+
+WhisperLocalModel = Literal["tiny", "small", "medium", "large", "large-v2"]
+WhisperRemoteModel = Literal["whisper-1"]
+WhisperModel = Union[WhisperLocalModel, WhisperRemoteModel]
+
+
+@component
+class WhisperTranscriber:
```

Comment:
I think it would simplify the logic quite a bit, also the testing.

If we go this way I think it would be better to open two separate PRs and ditch this one.